pkg/diagnostics/diagnosticCollector.go

```diff
@@ -0,0 +1,79 @@
+package diagnostics
+
+import (
+	"archive/zip"
+	"bytes"
+	"encoding/json"
+	"fmt"
+	"time"
+
+	server_utils "github.com/siglens/siglens/pkg/server/utils"
+	"github.com/valyala/fasthttp"
+)
+
+func CollectDiagnosticsAPI(ctx *fasthttp.RequestCtx) {
+	buf := new(bytes.Buffer)
+	zipWriter := zip.NewWriter(buf)
+
+	req := fasthttp.AcquireRequest()
+	resp := fasthttp.AcquireResponse()
+	defer fasthttp.ReleaseRequest(req)
+	defer fasthttp.ReleaseResponse(resp)
+
+	serverAddr := ctx.URI().Host()
+	req.SetRequestURI(fmt.Sprintf("http://%s%s/clusterStats", serverAddr, server_utils.API_PREFIX))
```

Comment:
As a start and as a framework this is great.
Now that the base stuff is in, we should not have this api (`/api/collect-diagonostics`) not make another http api call. This function should actually directly call the functions which were handling the `/api/clusterStats`  api.